File: android/VideoView/app/src/main/java/io/streamroot/dna/videoview/VideoViewPlayerInteractor.java
package io.streamroot.dna.videoview;

import android.widget.VideoView;

import org.jetbrains.annotations.NotNull;

import io.streamroot.dna.core.PlayerInteractor;
import io.streamroot.dna.core.TimeRange;

import java.util.Collections;
import java.util.List;

final class VideoViewPlayerInteractor implements PlayerInteractor {
    private VideoView videoView;

    VideoViewPlayerInteractor(VideoView mVideoView) {
        videoView = mVideoView;
    }

    @Override
    public long playbackTime() {
        return videoView.getCurrentPosition();
    }

    @NotNull
    @Override
    public List<TimeRange> loadedTimeRanges() {
        return Collections.emptyList();
    }
}

File: android/VideoView/app/src/main/java/io/streamroot/dna/videoview/VideoViewPlayerInteractor.kt
package io.streamroot.dna.videoview

import android.widget.VideoView
import io.streamroot.dna.core.PlayerInteractor
import io.streamroot.dna.core.TimeRange
import java.util.Collections

class VideoViewPlayerInteractor(
        private val mVideoView: VideoView
) : PlayerInteractor{
    override fun loadedTimeRanges(): MutableList<TimeRange> {
        return Collections.emptyList()
    }

    override fun playbackTime(): Long {
        return mVideoView.currentPosition.toLong()
    }
}